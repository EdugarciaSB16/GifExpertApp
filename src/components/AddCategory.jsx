import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('')

    const handleChange = (e) => {
        setInputValue( e.target.value )
    }

    const handleSubmit = (e) => {
        e.preventDefault()

        if( inputValue.trim().length > 2 ){
            setCategories( prevState => {
                return [inputValue, ...prevState]
            })
            setInputValue('')
        }
    }


    return (
        <form onSubmit={handleSubmit}>
            <input 
                type="text" 
                placeholder="Introduce una categoria"
                value={ inputValue }
                onChange={ handleChange }
                />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;