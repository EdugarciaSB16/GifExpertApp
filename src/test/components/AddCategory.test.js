import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';
import React from 'react'


describe('Pruebas en el componente AddCategory', () => {

    const setCategories = () => {}

    test('Debe de retornar la categoria', () => {
        const wrapper = shallow(<AddCategory setCategories={ setCategories } />)
        expect( wrapper ).toMatchSnapshot()
    })

})