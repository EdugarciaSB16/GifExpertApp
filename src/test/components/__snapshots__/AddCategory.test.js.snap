// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory Debe de retornar la categoria 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Introduce una categoria"
    type="text"
    value=""
  />
</form>
`;
